### PR TITLE
[WIP] Feature/docker

### PR DESCRIPTION
**Pull request by Leeaandrob**
Adding docker in dev process [do not merge yet] #316

### CODE REVIEW

**Comment by jellegerbrandy** [comment]
Ok, I think this needs some instructions in the README file with instructions for developers on how this should be used:

1. Install docker
2. clone the repo
3. start the containers
4. how to commit and how to update to the latest version

**Comment by Leeaandrob** [comment]
Okay, @jellegerbrandy. I am writing first the basics of projects works together first then I will write the readme about the use of docker.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+RUN apt install yarn
+
+# install jq
+RUN JQ_URL="https://circle-downloads.s3.amazonaws.com/circleci-images/cache/linux-amd64/jq-latest" \
+  && curl --silent --show-error --location --fail --retry 3 --output /usr/bin/jq $JQ_URL \
+  && chmod +x /usr/bin/jq \
+  && jq --version
+
+# install java 8
+#
```
Comment (jellegerbrandy) [inline Dockerfile]: What do we need Java for?

Comment (jellegerbrandy) [reply Dockerfile]: (in general, it seems that we have  lots of software installed here that we do not need, can you try to clean it up a bit?)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+ && tar xvzf openssl-1.1.0.tar.gz \
+ && cd openssl-1.1.0 \
+ && ./config -Wl,--enable-new-dtags,-rpath,'$(LIBRPATH)' \
+ && make \
+ && make install \
+ && ln -s /usr/local/lib/libcrypto.so.1.1 /usr/lib/ \
+ && rm /usr/bin/openssl \
+ && ln -s /usr/local/ssl/bin/openssl /usr/bin/openssl
+
+# parity 1.9.0
```
Comment (jellegerbrandy) [inline Dockerfile]: this will go to a separate docker file, right?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      && curl --silent --show-error --location --fail --retry 3 --output /tmp/chromedriver_linux64.zip "http://chromedriver.storage.googleapis.com/$CHROMEDRIVER_RELEASE/chromedriver_linux64.zip" \
+      && cd /tmp \
+      && unzip chromedriver_linux64.zip \
+      && rm -rf chromedriver_linux64.zip \
+      && mv chromedriver /usr/local/bin/chromedriver \
+      && chmod +x /usr/local/bin/chromedriver \
+      && chromedriver --version
+
+# start xvfb automatically to avoid needing to express in circle.yml
+ENV DISPLAY :99
```
Comment (jellegerbrandy) [inline Dockerfile]: we can remove all the circleci stuff in the dev machines

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      context: .
+      dockerfile: Dockerfile-db
+    image: paratii-portal
+    command: >
+        bash -c "sleep 30 && npm run dev"
+  mongo:
+    container_name: mongo
+    image: mongo:latest
+    ports:
+        - "27017:27017"
```
Comment (jellegerbrandy) [inline docker-compose.yml]: still missing the parity docker.

File: Dockerfile (hunk 1 above)
Comment (Leeaandrob) [reply Dockerfile]: yes @jellegerbrandy you have right! I have some changes locally that I need to push.

File: Dockerfile (hunk 2 above)
Comment (Leeaandrob) [reply Dockerfile]: Yes. I am finishing the configuration the paratii-db first to do it @jellegerbrandy.

File: Dockerfile (hunk 3 above)
Comment (Leeaandrob) [reply Dockerfile]: Yes. @jellegerbrandy

File: docker-compose.yml (hunk 4 above)
Comment (Leeaandrob) [reply docker-compose.yml]: Added parity @jellegerbrandy

**Comment by jellegerbrandy** [comment]
@Leeaandrob , there are 100 changed files in this PR, and most of them have nothign to do with docker. What happened?

**Comment by Leeaandrob** [comment]
I did a rebase on my branch @jellegerbrandy to solve a fix related error with wallet @jellegerbrandy

File: .docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,37 @@
+FROM centos:latest
```
[the remark this answers is not in the document]
Comment (Leeaandrob) [reply .docker/Dockerfile]: Could we use the same OS that we are using in production mode? @jellegerbrandy

File: .docker/Dockerfile-db [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+# Use unicode
+RUN locale-gen C.UTF-8 || true
+ENV LANG=C.UTF-8
+
+RUN apt-get update \
+  && mkdir -p /usr/share/man/man1 \
+  && apt-get install -y \
+    git mercurial xvfb \
+    locales sudo openssh-client ca-certificates tar gzip parallel \
+    net-tools netcat unzip zip bzip2 gnupg curl wget build-essential curl wget
```
[the remark this answers is not in the document]
Comment (Leeaandrob) [reply .docker/Dockerfile-db]: yes, I will remove unused dependencies @jellegerbrandy

File: src/scripts/actions/ModalActions.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+/* @flow */
```
[the remark this answers is not in the document]
Comment (Leeaandrob) [reply src/scripts/actions/ModalActions.js]: I had made rebase now I am working in another PR without the rebase @jellegerbrandy 
https://github.com/Paratii-Video/paratii-portal/pull/384

File: DOCKER-README.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+
+## First installation:
+
+* Install docker
+* $ git clone https://github.com/Paratii-Video/paratii-portal
+* $ cd paratii-portal
+* Create a .env with LOCAL_IP=<your_local_ip> provided by:
+
+$ ifconfig | grep -Eo 'inet (addr:)?([0-9]*\.){3}[0-9]*' | grep -Eo '([0-9]*\.){3}[0-9]*' | grep -v '127.0.0.1'
```
[the remark this answers is not in the document]
Comment (Leeaandrob) [reply DOCKER-README.md]: Yes, you have absolutely right @jellegerbrandy but always that you change the network you need to change the .env with the new ip received by ```$ ifconfig | grep -Eo 'inet (addr:)?([0-9]*\.){3}[0-9]*' | grep -Eo '([0-9]*\.){3}[0-9]*' | grep -v '127.0.0.1'``` after this docker-compose up again @jellegerbrandy

File: DOCKER-README.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+In that case, you need to re-run `docker-compose up --build`
+
+### Testing
+
+The tests can be running outside of containers and see from your browser.
+
+To run unit tests:
+
+    $ yarn run unit-test
```
[the remark this answers is not in the document]
Comment (Leeaandrob) [reply DOCKER-README.md]: I will change this for ```npm```

[the remark this answers is not in the document]
Comment (Leeaandrob) [reply DOCKER-README.md]: Yes, you will need to do ```npm install``` to run the tests on your host machine to run the tests outside of the container. @jellegerbrandy

File: DOCKER-README.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+For run chimp tests:
+
+    $ yarn run chimp-test
```
[the remark this answers is not in the document]
Comment (Leeaandrob) [reply DOCKER-README.md]: Yes, you will need to do ```npm install``` to run the tests on your host machine to run the tests outside of the container. @jellegerbrandy

[the remark this answers is not in the document]
Comment (Leeaandrob) [reply DOCKER-README.md]: I will add this instruction to README too.

File: DOCKER-README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (Leeaandrob) [reply DOCKER-README.md]: I will add this instruction to README too.

File: .docker/Dockerfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (jellegerbrandy) [reply .docker/Dockerfile]: We should use the same OS for all machines everywhere. And at this point, i'd prefer debian.

**Pull request closed without merging**